src/net/packet.cc

```diff
@@ -137,6 +137,14 @@ std::ostream& operator<<(std::ostream& os, const packet& p) {
     return os;
 }
 
+packet packet::make(std::vector<temporary_buffer<char>> bufs) {
+    packet p(bufs.size());
+    for (auto&& buf : bufs) {
+        p = packet(std::move(p), std::move(buf));
+    }
```

Comment:
Is there no nicer way?

Reply:
Not today :D There's a bunch of "appending" constructors for packet:

```c++
    // append fragment (copying new fragment)
    packet(packet&& x, fragment frag);
    // append fragment (zero-copy)
    packet(packet&& x, fragment frag, deleter d);
    // append temporary_buffer (zero-copy)
    packet(packet&& x, temporary_buffer<char> buf);
```

The `packet(std::vector<temporary_buffer>)` one (or better `packet(std::span<temporary_buffer>)`) in the next version TODO list

Reply:
ok

Reply:
`net::packet(std::span<temporary_buffer>)` constructor in new version